src/components/GenericModal.vue

```diff
@@ -190,6 +190,14 @@ export default {
     closeModal() {
       this.isVisible = false;
     },
+    /**
+     * Select all modal's content
+     */
+    selectContent() {
```

Comment:
I like this approach better since it is more generic about the modal and not specific to code listings.

However, I think it would be even better if we could also avoid having to listen for the keyboard event and re-implementing the built-in behavior for actually selecting the text.

While thinking about this a bit, I'm wondering if we can simplify this PR to just applying the [`user-select: none`](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select) CSS style to everything that isn't the modal (the page background) when the modal is open. That way, we wouldn't need any JavaScript for keyboard events or actually doing the selection itself—essentially, the page just adds a CSS style to prevent selection only when the modal is visible.

What do you think @marinaaisa? (\cc @dobromir-hristov)

Reply:
I think that Safari does not respect `user-select: none` when you try to copy stuff. Its deff worth a shot though.

Reply:
As @dobromir-hristov mentioned, I tried `-webkit-user-select: none;` on `main` tag in Safari and it doesn't prevent it to copy the content inside.

@mportiz08, although I always prefer CSS solutions to JS ones, your proposal would break the principle of component isolation since it needs all the rest of html blocks in the page to have a certain CSS property.

In my opinion, this modal's feature should be scoped to the modal component, even if that means re-implementing the built-in behavior.
Thanks!

Reply:
> As @dobromir-hristov mentioned, I tried -webkit-user-select: none; on main tag in Safari and it doesn't prevent it to copy the content inside.

It is working for me when I try it, but maybe I'm missing something important.

> In my opinion, this modal's feature should be scoped to the modal component, even if that means re-implementing the built-in behavior.

I think that's generally a good guideline, but I feel like this desired behavior is an app-level feature where we're restricting selection to the background of the modal, so it seems reasonable to me that we could apply a CSS rule to a global element for that. The modal is kind of tricky due to the use of the portal. The solution you have here also depends on how the text is selected (it only works if done with the right keyboard shortcut), so it feels a little suboptimal but maybe that will end up being the only reasonable workaround.

Maybe I could try a quick prototype of what I had in mind, and we can see which strategy we prefer?

Reply:
Nevermind. I tried my little prototype with the below code, and it doesn't work completely like I was hoping (disabling selection works but re-enabling it just for the modal seems to re-enable it for the full page in Safari for some reason).

```diff
diff --git a/src/components/Tutorial.vue b/src/components/Tutorial.vue
index a028671..d504590 100644
--- a/src/components/Tutorial.vue
+++ b/src/components/Tutorial.vue
@@ -9,7 +9,7 @@
 -->
 
 <template>
-  <div class="tutorial">
+  <div class="tutorial" :class="dynamicClasses">
     <NavigationBar
       v-if="!isTargetIDE"
       :technology="metadata.category"
@@ -25,7 +25,7 @@
       />
       <BreakpointEmitter @change="handleBreakpointChange" />
     </main>
-    <PortalTarget name="modal-destination" multiple />
+    <PortalTarget name="modal-destination" multiple @change="onPortalChange" />
   </div>
 </template>
 
@@ -85,7 +85,11 @@ export default {
     'isTargetIDE',
     'store',
   ],
+  data: () => ({ hasPortalContent: false }),
   computed: {
+    dynamicClasses: ({ hasPortalContent }) => ({
+      unselectable: hasPortalContent,
+    }),
     heroSection() {
       return this.sections.find(({ kind }) => kind === 'hero');
     },
@@ -125,6 +129,9 @@ export default {
     handleCodeColorsChange(codeColors) {
       CodeThemeStore.updateCodeColors(codeColors);
     },
+    onPortalChange(hasContent) {
+      this.hasPortalContent = hasContent;
+    },
   },
   created() {
     this.store.reset();
@@ -150,5 +157,13 @@ export default {
 
 .tutorial {
   background-color: var(--colors-text-background, var(--color-tutorial-background));
+
+  &.unselectable {
+    -webkit-user-select: none;
+  }
+}
+
+/deep/ .modal-open {
+  -webkit-user-select: text;
 }
 </style>
```

Unless someone else spots something obviously wrong with what I tried, we will likely need to go with the workaround for overriding the default behavior for ctrl+a :cry: